test/support/ts-app/src/index.ts

```diff
@@ -17,4 +17,19 @@ app.get('/', async ctx => {
 
 app.get('/hello').to('foo#hello');
 
+app.get('/hello-name', async ctx => {
+  const name = (await ctx.params()).get('name');
+  await ctx.render({text: `Hello ${name}!`});
+});
+
+app.post('/data', async ctx => {
+  const body: any = await ctx.req.json();
+  await ctx.render({json : {greeting: `Hello ${body.name}`}});
```

Comment:
Extra whitespace after `json`. I guess we need an ESLint rule for this.

Reply:
We do have ESLint rules that should disallow this, seems they are not properly applied to the TypeScript tests.